pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +53,66 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
+	vo := VersionObj{nil, nil}
+
+	cvg := version.Get()
 	if cmdutil.GetFlagBool(cmd, "short") {
-		v = version.Get().GitVersion
+		cvg = version.Info{GitVersion: version.Get().GitVersion}
 	}
+	vo.ClientVersion = &cvg
 
-	fmt.Fprintf(out, "Client Version: %s\n", v)
-	if cmdutil.GetFlagBool(cmd, "client") {
-		return nil
-	}
+	if !cmdutil.GetFlagBool(cmd, "client") {
+		clientSet, err := f.ClientSet()
+		if err != nil {
+			return err
+		}
 
-	clientset, err := f.ClientSet()
-	if err != nil {
-		return err
-	}
+		serverVersion, err := clientSet.Discovery().ServerVersion()
+		if err != nil {
+			return err
+		}
 
-	serverVersion, err := clientset.Discovery().ServerVersion()
-	if err != nil {
-		return err
+		sv := serverVersion
+		if cmdutil.GetFlagBool(cmd, "short") {
+			sv = &version.Info{GitVersion: serverVersion.GitVersion}
```

Comment:
Same, don't truncate data here. Truncate in short mode only in the human readable case